lib/common/procfs.c

```diff
@@ -140,3 +140,35 @@ crm_procfs_pid_of(const char *name)
     closedir(dp);
     return pid;
 }
+
+/*!
+ * \internal
+ * \brief Calculate number of logical CPU cores from procfs
+ *
+ * \param[out] cores  Where to store result
+ *
+ * \return pcmk_ok on success, -errno otherwise
+ */
+int
+crm_procfs_num_cores(int *cores)
+{
+    FILE *stream = NULL;
+
+    CRM_CHECK(cores != NULL, return -EINVAL);
+    *cores = 0;
+    stream = fopen("/proc/cpuinfo", "r");
+
+    if (stream == NULL) {
+        return -errno;
+    } else {
+        char buffer[256];
+
+        while (fgets(buffer, sizeof(buffer), stream)) {
+            if (strncmp(buffer, "processor", 9) == 0) {
+                ++*cores;
+            }
+        }
```

Comment:
While at it, I know it's just copy-paste, but this is pretty naive way to look for a _contiguous_ string in a _contiguous_ block because of parcelling the windows.